src/resources/assets/js/geo/components/filterByLocationComponent.vue

```diff
@@ -33,17 +26,22 @@ export default {
     data() {
         return {
             selectedItem: null,
-            trigger: false,
+            showModal: false,
         };
     },
     methods: {
-        submit() {
-            // always trigger rule-refresh in geoFIlter.js (in case the rule has already been applied)
-            // not regarding whether sessionStorage-data has been changed or not 
-            window.dispatchEvent(customEvent);
-            // selectedItem is always null, so geoFilter-rule can only be added once
+        submit(key) {
+            this.hideModal();
+            // pass the array of selected IDs as JSON-string in order to work properly in the hasRule()-check later on.
+            // hasRule() function checks whether rule-attributes diverge between existing and new rules
+            // e.g. newRule.data === oldRule.data (does not assert equality for array- or object-comparison)
+            let ids = JSON.parse(sessionStorage.getItem(key)) || [];
+            this.selectedItem = JSON.stringify(ids);
```

Comment:
You could also update the biigle/core code to handle the array equality check correctly.

Reply:
pull request [here](https://github.com/biigle/core/pull/713#issue-2026192608)